src/services/permissions-checker.js

```diff
@@ -304,7 +158,7 @@ class PermissionsChecker {
   async checkPermissions(collectionName, permissionName, permissionInfos) {
     // TODO IN NEXT PR: Distinguish collectionsPermissionExpired and scopePermissionExpired for
     // rolesACL format and retrieve only scope when needed.
-    if (this._isPermissionExpired(permissionName)) {
+    if (PermissionsCache.isPermissionExpired(this.renderingId, permissionName)) {
```

Comment:
Well the `isPermissionExpired` should actually be a private function. And here you should have only one function to get the permissions
```js
const permissions = await PermissionsGetter.getPermissionsForRendering(this.renderingId);
```

Reply:
This get permissions should return **ALWAYS** the same format and normally you should have no need to call `PermissionsGetter` anywhere else in this file 😉

Reply:
Ok for the same format of the `PermissionsGetter.getPermissions`.
However, `isPermissionExpired` should not be a private function to me (see my comment below).